src/vs/workbench/contrib/tasks/browser/task.contribution.ts

```diff
@@ -40,7 +40,7 @@ import { TaskDefinitionRegistry } from 'vs/workbench/contrib/tasks/common/taskDe
 import { TerminalMenuBarGroup } from 'vs/workbench/contrib/terminal/browser/terminalMenus';
 import { isString } from 'vs/base/common/types';
 
-const SHOW_TASKS_COMMANDS_CONTEXT = ContextKeyExpr.or(ShellExecutionSupportedContext, ProcessExecutionSupportedContext);
+const SHOW_TASKS_COMMANDS_CONTEXT = ContextKeyExpr.and(ShellExecutionSupportedContext, ProcessExecutionSupportedContext);
```

Comment:
@alexr00 this looked a little off to me, but it does align with the `&&` in `_jsonTasksSupported`?

Reply:
This context is used to determine when to show the task commands. If either shell or process tasks are supported, then the task commands should be shown. I don't really expect that only one (either shell or process, but not both) will ever happen as things are currently written however.

In `_jsonTasksSupported` we are checking to see if tasks.json tasks should be supported. In this case, we only say they're supported when both shell and process tasks are supported because we only have 1 json schema, which allows both shell and process.

Reply:
Since this is currently not expected state it doesn't really matter if you want to change this context.